plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedOnceRetentionStrategy.java

```diff
@@ -83,6 +85,13 @@ private void done(Executor executor) {
     @SuppressFBWarnings(value="SE_BAD_FIELD", justification="not a real Callable")
     private void done(final AbstractCloudComputer<?> c) {
         c.setAcceptingTasks(false); // just in case
+        if (c.isOffline() && c.getOfflineCause() instanceof OfflineCause.UserCause) {
+            if (!tempOffline) {
```

Comment:
Move this fast test before the offline cause tests.

Reply:
I don't understand here... tempOffline test is used only for deduction if log msg should be generated or not, otherwise test for 'offline' is used if the slave should be released or not yet. The time spent in this logic and in the logic you suggest looks close to me. Probably I'm missing sth. here.

Reply:
Right, I have missed the `return`.